static/app/views/settings/project/projectKeys/projectKeyCredentials.tsx

```diff
@@ -10,210 +10,186 @@ import space from 'sentry/styles/space';
 import getDynamicText from 'sentry/utils/getDynamicText';
 import {ProjectKey} from 'sentry/views/settings/project/projectKeys/types';
 
-const DEFAULT_PROPS = {
-  showDsn: true,
-  showDsnPublic: true,
-  showSecurityEndpoint: true,
-  showMinidump: true,
-  showUnreal: true,
-  showPublicKey: false,
-  showSecretKey: false,
-  showProjectId: false,
-};
-
 type Props = {
   data: ProjectKey;
   projectId: string;
-} & typeof DEFAULT_PROPS;
-
-type State = {
-  showDeprecatedDsn: boolean;
+  showDsn?: boolean;
+  showDsnPublic?: boolean;
+  showMinidump?: boolean;
+  showProjectId?: boolean;
+  showPublicKey?: boolean;
+  showSecretKey?: boolean;
+  showSecurityEndpoint?: boolean;
+  showUnreal?: boolean;
 };
 
-class ProjectKeyCredentials extends Component<Props, State> {
-  static defaultProps = DEFAULT_PROPS;
-
-  state: State = {
-    showDeprecatedDsn: false,
-  };
-
-  toggleDeprecatedDsn = () => {
-    this.setState(state => ({
-      showDeprecatedDsn: !state.showDeprecatedDsn,
-    }));
-  };
-
-  render() {
-    const {showDeprecatedDsn} = this.state;
-    const {
-      projectId,
-      data,
-      showDsn,
-      showDsnPublic,
-      showSecurityEndpoint,
-      showMinidump,
-      showUnreal,
-      showPublicKey,
-      showSecretKey,
-      showProjectId,
-    } = this.props;
-
-    return (
-      <Fragment>
-        {showDsnPublic && (
-          <Field
-            label={t('DSN')}
-            inline={false}
-            flexibleControlStateSize
-            help={tct('The DSN tells the SDK where to send the events to. [link]', {
-              link: showDsn ? (
-                <Link to="" onClick={this.toggleDeprecatedDsn}>
-                  {showDeprecatedDsn
-                    ? t('Hide deprecated DSN')
-                    : t('Show deprecated DSN')}
-                </Link>
-              ) : null,
+function ProjectKeyCredentials({
+  data,
+  projectId,
+  showDsn = true,
+  showDsnPublic = true,
+  showMinidump = true,
+  showProjectId = false,
+  showPublicKey = false,
+  showSecretKey = false,
+  showSecurityEndpoint = true,
+  showUnreal = true,
+}: Props) {
+  const [showDeprecatedDsn, setShowDeprecatedDsn] = useState(false);
+
+  return (
+    <Fragment>
+      {showDsnPublic && (
+        <Field
+          label={t('DSN')}
+          inline={false}
+          flexibleControlStateSize
+          help={tct('The DSN tells the SDK where to send the events to. [link]', {
+            link: showDsn ? (
+              <Link to="" onClick={() => setShowDeprecatedDsn(curr => !curr)}>
```

Comment:
```suggestion
              <Link to="" onClick={() => setShowDeprecatedDsn(!showDeprecatedDsn)}>
```

What does `curr` mean?

Reply:
That's the current value of the state. It's similar to `this.setState(currentState => ...)`

You can read more here https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often